python/pyarrow/tests/test_fs.py

```diff
@@ -23,39 +25,276 @@
+
+        self.impl = SubTreeFileSystem(
+            prefix_absolute.as_posix(),
+            LocalFileSystem()
+        )
+        self.prefix = prefix
+        self.tempdir = tempdir
+
+    def pathpair(self, p):
+        path_for_wrapper = str(self.tempdir / self.prefix / p)
+        path_for_impl = p
+        return (path_for_wrapper, path_for_impl)
+
+
+class S3Wrapper(FileSystemWrapper):
+
+    allow_move_dir = False
+
+    def __init__(self, minio_client, bucket='test-bucket', **kwargs):
+        from pyarrow.fs import S3FileSystem
+        self.impl = S3FileSystem(**kwargs)
+        self.client = minio_client
+        self.bucket = bucket
+
+    def pathpair(self, p):
+        path_for_wrapper = p
+        path_for_impl = '/'.join([self.bucket, p])
+        return (path_for_wrapper, path_for_impl)
+
+    def touch(self, p):
+        self.client.put_object(
+            bucket_name=self.bucket,
+            object_name=p.rstrip('/'),
+            data=io.BytesIO(b''),
+            length=0
+        )
+
+    def unlink(self, p):
+        self.client.remove_object(
+            bucket_name=self.bucket,
+            object_name=p.rstrip('/')
+        )
+
+    def rmdir(self, p):
+        if not p.endswith('/'):
+            p += '/'
+        self.client.remove_object(
+            bucket_name=self.bucket,
+            object_name=p
+        )
+
+    def mkdir(self, p):
+        if not p.endswith('/'):
+            p += '/'
+        self.client.put_object(
+            bucket_name=self.bucket,
+            object_name=p,
+            data=io.BytesIO(b''),
+            length=0
+        )
+
+    def iterdir(self, p):
+        if not p.endswith('/'):
+            p += '/'
+        objs = self.client.list_objects(
+            bucket_name=self.bucket,
+            prefix=p,
+            recursive=False
+        )
+        for obj in objs:
+            yield (obj.object_name, obj.is_dir)
+
+    def exists(self, p):
+        from minio.error import NoSuchKey, NoSuchBucket
+        try:
+            self.client.get_object(
+                bucket_name=self.bucket,
+                object_name=p
+            )
+        except (NoSuchBucket, NoSuchKey):
+            return False
+        else:
+            return True
+
+    def mtime(self, p):
+        stat = self.client.stat_object(
+            bucket_name=self.bucket,
+            object_name=p
+        )
+        ts = calendar.timegm(stat.last_modified)
+        return datetime.utcfromtimestamp(ts)
+
+    def write_bytes(self, p, data):
+        assert not p.endswith('/')
+        self.client.put_object(
+            bucket_name=self.bucket,
+            object_name=p,
+            data=io.BytesIO(data),
+            length=len(data)
+        )
+
+    def read_bytes(self, p):
+        assert not p.endswith('/')
+        data = self.client.get_object(
+            bucket_name=self.bucket,
+            object_name=p
+        )
+        return data.read()
+
+
+class SubTreeS3Wrapper(S3Wrapper):
```

Comment:
There is no need to test this.

Reply:
Why if we expose both `SubTreeFileSystem` and `S3FileSystem`?

Reply:
I've already set up the S3 testing suite with Minio, which also helps testing s3fs integration. 

Of course I can remove it from the pytest parametrization, but IMO it's better to have tests (it's quick).